### PR TITLE
Feature/Authentication, with Event creation and display

### PR DESCRIPTION
**Pull request by cliftondavies**
# Project: Associations
## Milestones:
- [x] Gitflow
- [x] Github Actions
- [x] Rubocop
- [x] Stylelint
- [x] .gitignore
- [x] Foreign keys have corresponding indexes
- [x] Unit tests for models associations and validations, using RSpec
- [x] Integrations tests for authentication and events creation, using RSpec and Capybara
- [x] README

### CODE REVIEW

**Comment by santiago-rodrig** [comment]
# Pull Request under review by @santiago-rodrig (TSE)

**Comment by cliftondavies** [comment]
Hi @santiago-rodrig :wave: Thanks for the review :+1:

I've provided a few screenshots below relating to the three changes you required.

 ### You are not providing any way to specify the event date at the event creation:

The [project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations) required that the form for creating an event contain a **description** field only, and none of the other Event attributes such as the **event date**, or **location**.

event_form_fields

The project specific requirements listed by Microverse doesn't state anything contrary to that.

### You are not listing the events at the events index view:

event_index_page

Above is a screenshot of the event index view from my side, showing a list of all events separated into upcoming and past events. Each event also displays a **description**, **date**, and **location**. These were populated from the rails console, as the project only required a **description** field in the event creation form.

### You are not listing the attended and going to attend events in the user's profile

user_show_page

This is a screenshot of a user's show view showing a list of their created events, events they are going to attend, and events they've attended.


Hope these help provide some clarity.

Thanks again, and happy coding :pray: :technologist:

**Comment by santiago-rodrig** [comment]
> Hi @santiago-rodrig 👋 Thanks for the review 👍
> 
> I've provided a few screenshots below relating to the three changes you required.
> 
> ### 1. You are not providing any way to specify the event date at the event creation:
> The [project](https://www.theodinproject.com/courses/ruby-on-rails/lessons/associations) required that the form for creating an event contain a **description** field only, and none of the other Event attributes such as the **event date**, or **location**.
> 
> event_form_fields
> 
> The project specific requirements listed by Microverse doesn't state anything contrary to that.
> 
> ### 2. You are not listing the events at the events index view:
> event_index_page
> 
> Above is a screenshot of the event index view from my side, showing a list of all events separated into upcoming and past events. Each event also displays a **description**, **date**, and **location**. These were populated from the rails console, as the project only required a **description** field in the event creation form.
> 
> ### 3. You are not listing the attended and going to attend events in the user's profile
> user_show_page
> 
> This is a screenshot of a user's show view showing a list of their created events, events they are going to attend, and events they've attended.
> 
> I hope these help provide some clarity.
> 
> Thanks again, and happy coding 🙏 🧑‍💻

Hi @cliftondavies, I want to apologize, it was a mistake from my part, I didn't read the instructions for the students, I'm sorry.

I'm sure that the next review will be approved.

File: app/views/events/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<h2>All Events</h2>
+
+<p>All Upcoming Events</p>
+<% Event.upcoming.each do |event| %>
```
Comment (mosesogwo) [inline app/views/events/index.html.erb:4]: Please retrieve this collection into a variable in your controller and only use the variable  here

File: app/views/events/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+<ul>
+  <li><%= event.description %></li>
+  <li><%= event.event_date %></li>
+  <li><%= event.location %></li>
+  <li><%= link_to 'View Event', event %></li>
+</ul>
+<% end %>
+
+<p>All Past Events</p>
+<% Event.past.each do |event| %>
```
Comment (mosesogwo) [inline app/views/events/index.html.erb:14]: Please retrieve this collection into a variable in your controller and only use the variable  here

**Comment by cliftondavies** [comment]
Hi @mosesogwo! :wave: 

I have completed the changes requested: :memo: 

#### Retrieved `Event.past` and `Event.upcoming` collections in Event index view, into `@past` and `@upcoming `variables in Event controller. :heavy_check_mark: 

#### Included fields for `event_date` and `location` in new event form :heavy_check_mark: 

Thanks for the review :pray:
Happy coding! :man_technologist:

**Comment by cliftondavies** [comment]
Hi @maelfosso, thanks for the review. :pray: 

I have implemented the RSVP feature, as requested. :memo: :heavy_check_mark: 

Happy coding! :woman_technologist:

**Comment by cliftondavies** [comment]
Thanks, @sumancrest0001!! 👍 🙏

**Pull request merged**